tests/manage/z_cluster/nodes/test_simultaneous_ocs_node_drains.py

```diff
@@ -0,0 +1,133 @@
+from ocs_ci.framework.testlib import (
+    tier4,
+    tier4b,
+    ManageTest,
+    ignore_leftovers,
+    skipif_bm,
+    skipif_lso,
+)
+from ocs_ci.helpers.sanity_helpers import Sanity
+
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(autouse=True)
+def teardown(request):
+    """
+    Tear down function
+
+    """
+
+    def finalizer():
+        """
+        Make sure that all cluster's nodes are in 'Ready' state and if not,
+        change them back to 'Ready' state by marking them as schedulable
+        """
+        scheduling_disabled_nodes = [
+            n.name
+            for n in get_node_objs()
+            if n.ocp.get_resource_status(n.name)
+            == constants.NODE_READY_SCHEDULING_DISABLED
+        ]
+        if scheduling_disabled_nodes:
+            schedule_nodes(scheduling_disabled_nodes)
+
+    request.addfinalizer(finalizer)
+
+
+@ignore_leftovers
+class TestSimultaneousOcsNodeDrains(ManageTest):
+    """"""
+
+    @pytest.fixture(autouse=True)
+    def init_sanity(self):
+        """
+        Initialize Sanity instance
+
+        """
+        self.sanity_helpers = Sanity()
+
+    @pytest.fixture(autouse=True)
+    def health_checker(self):
+        """
+        Check Ceph health
+
+        """
+        try:
+            ceph_health_check_base()
+        except CephHealthException as e:
+            # skip because ceph is not in good health
+            pytest.skip(str(e))
+
+    @tier4
+    @tier4b
+    @skipif_bm
+    @skipif_lso
+    def test_simultaneous_node_drains(
+        self,
+        pvc_factory,
+        pod_factory,
+        bucket_factory,
+        rgw_bucket_factory,
+    ):
+        """
+        - Check for OSD PDBs before drain
+        - Maintenance (mark as unscheduable and drain) 2 worker node with delay of 30 secs
+        - Drain will be completed on worker node A
+        - Drain will be pending on worker node B due to blocking PDBs
+        - Check the OSD PDBs
+        - Mark the node A as schedulable
+        - Let drain finish on Node B
+        - Mark the node B as schedulable
+        - Check cluster and Ceph health
+
+        """
+        # Validate OSD PDBs before drain operation
+        validate_osd_pdb()
+
+        # Get 2 worker nodes to drain
+        typed_nodes = get_nodes(node_type=constants.WORKER_LABEL, num_of_nodes=2)
+        assert typed_nodes, f"Failed to find worker nodes for the test"
+        node_A = typed_nodes[0].name
+        node_B = typed_nodes[1].name
+
+        # Drain Node A and validate blocking PDBs
+        drain_nodes([node_A])
+        validate_osd_pdb()
+
+        # Inducing delay between 2 drains
+        # Node-B drain expected to be in pending due to blocking PDBs
+        time.sleep(30)
+        drain_nodes([node_B])
+
+        # Mark the node-A back to schedulable and let drain finish in Node-B
+        schedule_nodes([node_A])
+
+        # Validate OSD PDBs
+        validate_osd_pdb()
```

Comment:
I am confused with the intent of this function now :) 
Are you trying to validate the OSD PDB or check whether we have any blocking PDBs or not?

Reply:
The intend is to check both. The function checks if it had created blocking PDBs or it has OSD PDBs and return bool value accordingly.

Reply:
don't we need to have assertion or raise exception if the bool value doesn't match where we used the function `validate_osd_pdb()` accordingly, for e.g if we didn't have blocking PDBs created??

Reply:
Done